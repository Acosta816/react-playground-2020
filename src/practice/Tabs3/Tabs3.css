body {
    display: flex;
    justify-content: center;
}

.avatar {
    border: 2px solid rgba(0, 132, 255, 0.87);
    width: 300px;
    height: auto;
}