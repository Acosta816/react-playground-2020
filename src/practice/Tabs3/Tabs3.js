import React from 'react';
import './Tabs3.css';


class Tabs3 extends React.Component {

    static defaultProps = {
        tabs: [{name: "potatoe"}]
    }

    constructor(props){
        super(props);
        this.state= {
            currentTab: this.props.tabs[0]
        }
    }

//================================================================================class-level functions BEGIN================================

    renderButtons = ()=> {
        const buttonsArray= this.props.tabs.map((tab,index)=> {
            return(
                <button type="button" key={index} onClick={(e)=>{this.handleButton(e)}}>
                    {tab.name}
                </button>
            )
        });

        return buttonsArray;

    }

    renderCurrentTab = ()=> {
        const {currentTab} = this.state;
        return (
            <div className="current-tab">
                <h1>{currentTab.name}</h1>
                <img className="avatar" src={currentTab.avatar} alt={currentTab.content}/>
            </div>
        )
    }

    handleButton =(e)=> {
        
        const currentTab = this.props.tabs.find(tab=> tab.name === e.target.innerText);
        console.log(currentTab);

        this.setState({
            currentTab
        })
    }

//================================================================================class-level functions END================================


    render() {

        return(
            <div>
                {this.renderButtons()}
                {this.renderCurrentTab()}
            </div>
        )
    }
}

export default Tabs3;