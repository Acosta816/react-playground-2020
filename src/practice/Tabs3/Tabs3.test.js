import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tabs3 from './Tabs3';


describe('TestSuite: Tabs3 component', ()=> {
    const tabs = [{name:'Sonic', content: 'Fastest thing Alive', avatar: "https://b7.pngbarn.com/png/547/724/sonic-the-hedgehog-2-sonic-3d-sonic-mania-sonic-cd-sonic-the-hedgehog-png-clip-art.png"}, {name:'Tails', content: 'Loyal and Hyper-intelligent Flying Fox', avatar: "https://vignette.wikia.nocookie.net/crystal-crossover/images/c/cb/ManiaTailsFront.png/revision/latest?cb=20180805022514"}, {name:'Amy', content: 'Sonics Biggest Fan. Hammer-weilding Pink Hedgehog', avatar: 'https://b7.pngbarn.com/png/319/82/amy-rose-sonic-colors-sonic-chaos-sonic-cd-knuckles-the-echidna-sonic-the-hedgehog-png-clip-art.png'}, {name:'Robotnik', content: 'Crazed Genius, obsessed with robotisizing all life and ridding the world of Sonic'}, {name:'Shadow', content: 'Sonics adversary. Created by Robotnik to become the ultimate lifeform'}, {name:'Sally', content: 'Rightful Princess of Mobius and one of the Leaders in the Freedom Fighters against Robotnik. Sonics Best friend.'}];

    it('SMOKE: renders without crashing', ()=> {
        const div = document.createElement('div');
        ReactDOM.render(<Tabs3 />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Snap1: renders empty given no tabs', ()=> {
        const wrapper = shallow(<Tabs3/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Snap2: renders first tab by default', ()=> {
        const wrapper = shallow(<Tabs3 tabs={tabs}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Snap3: closes first tab and opens 3rd tab', ()=> {
        const wrapper = shallow(<Tabs3 tabs={tabs} />);
        wrapper.find('button').at(2).simulate('click'); //tranversing the wrapper and clicking the button at index#2 (third button)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

})