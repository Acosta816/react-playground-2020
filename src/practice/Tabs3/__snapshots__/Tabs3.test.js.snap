// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuite: Tabs3 component Snap1: renders empty given no tabs 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
    type="button"
  >
    potatoe
  </button>
  <div
    className="current-tab"
  >
    <h1>
      potatoe
    </h1>
    <img
      className="avatar"
    />
  </div>
</div>
`;

exports[`TestSuite: Tabs3 component Snap2: renders first tab by default 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
    type="button"
  >
    Sonic
  </button>
  <button
    key="1"
    onClick={[Function]}
    type="button"
  >
    Tails
  </button>
  <button
    key="2"
    onClick={[Function]}
    type="button"
  >
    Amy
  </button>
  <button
    key="3"
    onClick={[Function]}
    type="button"
  >
    Robotnik
  </button>
  <button
    key="4"
    onClick={[Function]}
    type="button"
  >
    Shadow
  </button>
  <button
    key="5"
    onClick={[Function]}
    type="button"
  >
    Sally
  </button>
  <div
    className="current-tab"
  >
    <h1>
      Sonic
    </h1>
    <img
      alt="Fastest thing Alive"
      className="avatar"
      src="https://b7.pngbarn.com/png/547/724/sonic-the-hedgehog-2-sonic-3d-sonic-mania-sonic-cd-sonic-the-hedgehog-png-clip-art.png"
    />
  </div>
</div>
`;
